src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/sender/LogListener.java

```diff
@@ -0,0 +1,42 @@
+package com.sumologic.jenkins.jenkinssumologicplugin.sender;
+
+import com.sumologic.jenkins.jenkinssumologicplugin.PluginDescriptorImpl;
+import com.sumologic.jenkins.jenkinssumologicplugin.integration.SearchAction;
+import hudson.Extension;
+import hudson.console.ConsoleLogFilter;
+import hudson.model.AbstractBuild;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * This class intercepts console output stream for every jenkins build and decorates it with additional functionality.
+ * Depending on configuration it can add two functionalities:
+ * 1) Add consistent timestamps to all output lines.
+ * 2) Forward all output logs to Sumo Logic.
```

Comment:
We can't piggy back. Jenkins timestamps are optional, just have hours.

Currently the plugin is designed as you proposed. We always add timestamp in logs sent to Sumo, but it's optional everywhere else.